lib/backends/sdc/containers.js

```diff
@@ -559,60 +525,302 @@ function addRulesToFWAPI(opts, rules, payload, callback) {
+ * see DOCKER-705), we will also attach to configured network pool
+ * (typically the external/public pool).
+ */
+function defaultFabricNetwork(opts, container, payload, callback) {
+    var dc = opts.app.config.datacenterName;
+    var log = opts.log;
+    var requireExternal = false;
+
+    if (publishingPorts(container) || opts.fabricRequireExternal) {
+        requireExternal = true;
+    }
+
+    log.debug({ requireExternal: requireExternal },
+        'Networks: using default fabric network');
+
+    opts.app.ufds.getDcLocalConfig(opts.account.uuid, dc, function (err, conf) {
+        log.debug({err: err, conf: conf, account: opts.account.uuid},
+            'Networks: get DC local config');
+
+        if (err || !conf || !conf.defaultnetwork) {
+            callback(errors.ufdsErrorWrap(err,
+                'Networks: could not get default network'));
             return;
         }
 
-        networks.forEach(function (n) {
-            if (!external_net
-                && n.name === netName) {
+        payload.networks = [ {uuid: conf.defaultnetwork} ];
+
+        if (requireExternal) {
+            payload.networks.push({uuid: opts.config.overlay.externalPool});
+        }
+
+        payload.networks[payload.networks.length - 1].primary = true;
+        return callback();
+    });
+}
+
+/*
+ * When fabrics are configured and a specific network name is supplied,
+ * we will find and use the specified network.
+ */
+function namedNetwork(opts, container, payload, callback) {
+
+    // need to search on networks by: name, fabric-true, owner_uuid
+    var log = opts.log;
+    var query = container.HostConfig.NetworkMode;
+    var napi = getNapiClient(opts.config.napi);
 
-                external_net = n.uuid;
+    vasync.parallel({
+        funcs: [
+            function byName(cb) {
+                var listParams = {
+                    name: query,
+                    fabric: true,
+                    owner_uuid: opts.account.uuid,
+                    limit: 2
+                };
+
+                log.debug({ listParams: listParams },
+                    format('Networks: searching for network %s',
+                        listParams.name));
+
+
+                napi.listNetworks(listParams,
+                    { headers: {'x-request-id': opts.req_id }}, cb);
+            },
+            function byDockerId(cb) {
+                // to search for docker ids, when NAPI only retains UUIDs, we
+                // rely on the 'double uuid' convention (see
+                // utils.networkUuidToDockerId()). To perform the search,
+                // we truncate the provided string to 8 chars, perform a
+                // wildcard search on networks, and post-filter the results to
+                // find more-complete matches than the at-most-8-char prefix
+                // might find.
+
+                var queryString = utils.ldapEscape(query.substr(0, 8)) + '*';
```

Comment:
Would `uuidSearchPattern` or something along those lines be a better name instead of `queryString`? It seems this variable does not represent the whole querystring of the following `listNetworks` call/request.